core/src/main/java/org/radargun/config/Cluster.java

```diff
@@ -53,23 +53,27 @@ public int getSize() {
    }
 
    public Group getGroup(int slaveIndex) {
-      int index = slaveIndex;
-      for (Group g : groups) {
-         if (index < g.size) {
-            return g;
+      if (slaveIndex != -1) {
```

Comment:
-1 is just one invalid value (or value with meaning of 'unset' value), but as you already added a check, you should check all 'unset' values. But this is just a nitpicking, I'll integrate this soonish.